automation/src/test/scala/org/broadinstitute/dsde/test/api/SamApiSpec.scala

```diff
@@ -401,26 +401,115 @@ class SamApiSpec extends FreeSpec with BillingFixtures with Matchers with ScalaF
         5.minutes, 5.seconds)
 
       // Change the membership of the admin policy to include users 1 and 2
-      Sam.user.setPolicyMembers(managedGroupId, adminPolicyName, Set(user1.email, user2.email))(user1AuthToken)
+      Sam.user.setManagedGroupPolicyMembers(managedGroupId, adminPolicyName, Set(user1.email, user2.email))(user1AuthToken)
       awaitAssert(
         Await.result(googleDirectoryDAO.listGroupMembers(policyEmail.head), 5.minutes)
           .getOrElse(Set.empty) should contain theSameElementsAs Set(user1Proxy.value, user2Proxy.value),
         5.minutes, 5.seconds)
 
       // Add user 3 to the admin policy
-      Sam.user.addUserToPolicy(managedGroupId, adminPolicyName, user3.email)(user1AuthToken)
+      Sam.user.addUserToManagedGroupPolicy(managedGroupId, adminPolicyName, user3.email)(user1AuthToken)
       awaitAssert(
         Await.result(googleDirectoryDAO.listGroupMembers(policyEmail.head), 5.minutes)
           .getOrElse(Set.empty) should contain theSameElementsAs Set(user1Proxy.value, user2Proxy.value, user3Proxy.value),
         5.minutes, 5.seconds)
 
       // Remove user 2 from the admin policy
-      Sam.user.removeUserFromPolicy(managedGroupId, adminPolicyName, user2.email)(user1AuthToken)
+      Sam.user.removeUserFromManagedGroupPolicy(managedGroupId, adminPolicyName, user2.email)(user1AuthToken)
       awaitAssert(
         Await.result(googleDirectoryDAO.listGroupMembers(policyEmail.head), 5.minutes)
           .getOrElse(Set.empty) should contain theSameElementsAs Set(user1Proxy.value, user3Proxy.value),
         5.minutes, 5.seconds)
     }
+
+    "should only synchronize the intersection group for policies constrained by auth domains" in {
+      implicit val ec: ExecutionContextExecutor = ExecutionContext.global
+      val authDomainId = UUID.randomUUID.toString
+      val Seq(inPolicyUser: Credentials, inAuthDomainUser: Credentials, inBothUser: Credentials) = UserPool.chooseStudents(3)
+      val inBothUserAuthToken = inBothUser.makeAuthToken()
+      val Seq(inAuthDomainUserProxy: WorkbenchEmail, inBothUserProxy: WorkbenchEmail) = Seq(inAuthDomainUser, inBothUser).map {
+        user => Sam.user.proxyGroup(user.email)(inBothUserAuthToken)
+      }
+
+      Sam.user.createGroup(authDomainId)(inBothUserAuthToken)
+      register cleanUp Sam.user.deleteGroup(authDomainId)(inBothUserAuthToken)
+
+      val authDomainPolicies = Sam.user.listResourcePolicies("managed-group", authDomainId)(inBothUserAuthToken)
+      val authDomainAdminEmail = for {
+        policy <- authDomainPolicies if policy.policy.memberEmails.nonEmpty
+      } yield {
+        policy.email
+      }
+      assert(authDomainAdminEmail.size == 1)
+
+      awaitAssert(
+        Await.result(googleDirectoryDAO.listGroupMembers(authDomainAdminEmail.head), 5.minutes)
```

Comment:
we talked about moving away from `Await`s in tests in favour of scalatest's `FutureSupport`, but I don't know if a) we ever did in Sam b) if it ended up being dangerous.